compendium/ServiceComponent/test/TestCompInst_RefOrder.cpp

```diff
@@ -111,9 +123,10 @@ namespace
         ComponentInstanceImpl<TestServiceImpl1,
                               std::tuple<>,
                               std::true_type,
-                              ServiceDependency2,
-                              ServiceDependency1>
-            compInstance(binders); // compile error
+                              std::shared_ptr<ServiceDependency2>,
+                              std::vector<std::shared_ptr<ServiceDependency3>>,
+                              std::shared_ptr<ServiceDependency1>>
+            compInstance({ "bar", "baz", "foo" }, binders); // compile error
```

Comment:
This test, along with `TestCompInst_NoInjectNoDefaultCtor.cpp` and others, requires manual removal of the `#ifdef NEVER` and re-compilation to test. This tests needs to run automatically and run as part of the overall build and test of this project. We can use CMake to compile these cpp files as compile-time tests (with the expectation they will fail compilation) that don't cause the build to fail on compiler error. See https://stackoverflow.com/a/50665823 for information on how to do this. This use of CMake to run compile time tests needs to be applied to `TestCompInst_InvalidInheritance.cpp`, `TestCompInst_NoDepsNoDefaultCtor.cpp`, `TestCompInst_NoInjectNoDefaultCtor.cpp`, and `TestCompInst_RefCount.cpp` as well.